compose/ui/ui/src/uikitInstrumentedTest/kotlin/androidx/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -43,18 +40,37 @@ import platform.Foundation.NSRunLoop
 import platform.Foundation.NSValue
 import platform.Foundation.dateWithTimeIntervalSinceNow
 import platform.Foundation.runUntilDate
+import platform.UIKit.UIApplication
 import platform.UIKit.UIKeyboardAnimationCurveUserInfoKey
 import platform.UIKit.UIKeyboardAnimationDurationUserInfoKey
 import platform.UIKit.UIKeyboardFrameEndUserInfoKey
 import platform.UIKit.UIKeyboardWillChangeFrameNotification
+import platform.UIKit.UIKeyboardWillHideNotification
 import platform.UIKit.UIKeyboardWillShowNotification
 import platform.UIKit.UIScreen
 import platform.UIKit.UIWindow
 import platform.UIKit.valueWithCGRect
 
+/**
+ * Sets up the test environment for iOS instrumented tests, runs the given [test][testBlock]
+ * and then tears down the test environment. Use the methods on [UIKitInstrumentedTest]
+ * in the test to find compose content and make assertions on it.
+ *
+ * Tests run without [UIApplication] and Application Delegate are created.
+ * Because of this, there are several known issues:
+ * - Rendering calls do not take effect. Also, it's not possible to take screenshots of the views.
+ * - The [becomeFirstResponder()] method does not display the keyboard on the first call.
+ *   Use the dedicated keyboard methods of the [UIKitInstrumentedTest]
+ *   class to manipulate the keyboard appearance.
+ *
+ * @param [testBlock] The test function.
+ */
 internal fun runUIKitInstrumentedTest(
     testBlock: UIKitInstrumentedTest.() -> Unit
-) = with(UIKitInstrumentedTest(), testBlock)
+) = with(UIKitInstrumentedTest()) {
+    testBlock()
+    tearDown()
```

Comment:
Maybe in `finally`?

Reply:
`tearDown()` is a standard function name for such cleanup operations

Reply:
Thanks, that may work!